jsapp/js/account/plans/addOnList.component.tsx

```diff
@@ -92,6 +108,48 @@ const AddOnList = (props: {
       .catch(handleCheckoutError);
   };
 
+  const renderUpdateBadge = (price: Price) => {
+    if (!(subscriptionUpdate && isSubscribedAddOnPrice(price))) {
+      return null;
+    }
+
+    let color: BadgeColor;
+    let label;
+
+    if (subscriptionUpdate.type === SubscriptionChangeType.CANCELLATION) {
+      color = 'light-red';
+      label = t('Ends on ##cancel_date##').replace(
+        '##cancel_date##',
+        formatDate(subscriptionUpdate.date)
+      );
+    } else if (subscriptionUpdate.type === SubscriptionChangeType.RENEWAL) {
+      color = 'light-blue';
+      label = t('Renews on ##renewal_date##').replace(
+        '##renewal_date##',
+        formatDate(subscriptionUpdate.date)
+      );
+    } else if (
+      subscriptionUpdate.type === SubscriptionChangeType.PRODUCT_CHANGE
+    ) {
+      if (currentPlan?.items[0].price.product.id === price.product) {
+        color = 'light-amber';
+        label = t('Ends on ##end_date##').replace(
+          '##end_date##',
+          formatDate(subscriptionUpdate.date)
+        );
+      } else {
+        color = 'light-teal';
+        label = t('Starts on ##start_date##').replace(
+          '##start_date##',
+          formatDate(subscriptionUpdate.date)
+        );
+      }
+    } else {
+      return null;
+    }
```

Comment:
This might read a bit better as a `switch`/`case`, since it's just checking a simple strict equality.